### PR TITLE
Add `trunkls` solver structure

### PR DESCRIPTION
**Pull request by tmigot**
This is a basic structure inspired by `trunk`, I tried to keep it simple to ease the reading.
Further optimization will definitely be possible. #131

### CODE REVIEW

File: src/trunkls.jl [hunk 1]
```diff
@@ -48,55 +50,120 @@ x0 = [-1.2; 1.0]
...
   if !unconstrained(nlp)
     error("trunk should only be called for unconstrained problems. Try tron instead")
   end
 
+  reset!(stats)
   start_time = time()
   elapsed_time = 0.0
 
+  solver.x .= x
+  x = solver.x
```
Comment (dpo) [inline src/trunkls.jl:143]:
```suggestion
```

**Review by dpo — CHANGES_REQUESTED** [review]
will the callback be in a separate pr?

Maybe allocation tests are in order.

File: src/trunkls.jl (hunk 1 above)
Comment (dpo) [reply src/trunkls.jl:143]: Sorry, just realize that this line is in fact necessary. Please ignore this comment.

**Comment by dpo** [comment]
See #135 though

**Comment by tmigot** [comment]
I suggest we handle #135 , callbacks and allocations tests in separate PRs.

**Pull request merged**